mautic/mautic
# Campaign action to permanently delete a contact added

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/161
| Related developer documentation PR URL | N
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/2562, partly https://github.com/mautic/mautic/issues/1172
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds a new campaign action which will delete a lead permanently. It can be used in classic campaign flows or as a way to delete all contacts in a segment with simple campaign. The upside is it uses the background batch process so it'll work for big numbers of contacts. Downside is it will delete all information about the contact including the statistics, so it won't record how many times this action was triggered.

#### Steps to test this PR:
1. Add the **Delete contact** action to a campaign. Make sure it contains some contacts. 
2. Make sure that no other campaign event can be added after the Delete contact action in the Campaign builder.
3. Run the `app/console mautic:campaign:trigger` command.
4. Check if the right contacts were deleted.
